### PR TITLE
Form block: Update icon property on all form variations

### LINKED ISSUE #17089: Form Block: icon color changes broke the main Form view
In #15311, we changed the color of all block icons. This appears to have broken the form block.

#### Steps to reproduce the issue

1. Start with a site running Jetpack's `master`.
2. Go to Posts > Add New
3. Add a Form block (not a form block variation, just the main form block).
4. You'll get the following error:

image

https://reactjs.org/docs/error-decoder.html/?invariant=31&args[]=object%20with%20keys%20%7Bsrc%2C%20foreground%7D&args[]=

```
Objects are not valid as a React child (found: object with keys {src, foreground}). If you meant to render a collection of children, use an array instead.
```

### PR DESCRIPTION
**Pull request by keoshi**
- Fixes a bug introduced in https://github.com/Automattic/jetpack/pull/15311.
- Closes https://github.com/Automattic/jetpack/issues/17089.

#### Changes proposed in this Pull Request:

Simplifies the icon property to have just a single value instead of an object where `src` and `foreground` were being passed onto. This formatting is not supported, nor are color changes in the variations selections (yet?).

#### Testing instructions:

* Go to Posts > New Post.
* Add the Form block.
* Confirm no errors are thrown, validate that the variation selection works, as well as all variations.

#### Screenshots

Before

image

After

image

#### Proposed changelog entry for your changes:

* None

### CODE REVIEW

**Review by jeherve — APPROVED** [review]

**Review by jeherve — APPROVED** [review]
Slack discussion: p1599472314418000-slack-C0D96691V

tl;dr: this may not be the best scenario (all Jetpack blocks have green icons, except for all form variations), but it seems to be the best we can do at the moment.

**Review by jeherve — CHANGES_REQUESTED** [review]
Quick change: since we don't use `getIconColor` anymore in that file, could you remove the import?

**Review by jeherve — CHANGES_REQUESTED** [review]
Let's maybe take that opportunity to revert the changes in `extensions/blocks/send-a-message/variations.js` as well. That main "Send a message" picker isn't displayed in the block picker right now since we only have one variation so far, but we may choose to display it in the future.

**Comment by keoshi** [comment]
Great catches, @jeherve ! Done both.

**Review by jeherve — APPROVED** [review]

**Review by scottsweb — APPROVED** [review]
Let's ship it and then I will see if I can further debug what is happening
